src/__tests__/helpers.js

```diff
@@ -25,9 +25,22 @@ describe('window retrieval throws when given something other than a node', () =>
       `It looks like you passed an Array instead of a DOM node. Did you do something like \`fireEvent.click(screen.getAllBy...\` when you meant to use a \`getBy\` query \`fireEvent.click(screen.getBy...\`?`,
     )
   })
+  test('HTMLElement as node', () => {
+    const elem = document.createElement('div')
+    Object.defineProperty(elem, 'ownerDocument', {
+      get: function get() {
+        return null
+      },
+    })
```

Comment:
This is basically mocking `document`. I'm not sure what this particular pattern is simulating.

Reply:
It's a workaround to skip over the [`node.ownerDocument && node.ownerDocument.defaultView` condition](https://github.com/testing-library/dom-testing-library/blob/main/src/helpers.js#L30). I don't know what we could do otherwise, because the created element is attached to the `document`. 
What I've tried as alternatives: removing the created element from the document, assigning the `ownerDocument` or `defaultView` to `null`.

Reply:
> What I've tried as alternatives: removing the created element from the document,

What does "removing from the document" mean? Note that an element always has `ownerDocument`. Whether it's connected (`element.isConnected`) or not, does not matter.

Reply:
> Note that an element always has ownerDocument. Whether it's connected (element.isConnected) or not, does not matter

That's good to know, that's where I got it wrong. I assumed the issue had an issue with `render(<input />)`.